### PR TITLE
Added close_pull_request event trigger, to support cleanups of PR deployments.

### PR DESCRIPTION
**Pull request by fhp**
I have added the `close_pull_request` event trigger to make it possible to cleanup a PR environment with the github hooks.

### CODE REVIEW

**Comment by bradrydzewski** [comment]
Please see the _Avoid Breaking Changes_ section of [this comment](https://github.com/drone/drone/issues/1878#issuecomment-287598789).

File: core/event.go [hunk 1]
```diff
@@ -16,11 +16,12 @@ package core
...
-	EventPush        = "push"
-	EventPullRequest = "pull_request"
-	EventTag         = "tag"
-	EventPromote     = "promote"
-	EventRollback    = "rollback"
+	EventCron             = "cron"
+	EventCustom           = "custom"
+	EventPush             = "push"
+	EventPullRequest      = "pull_request"
+	EventClosePullRequest = "close_pull_request"
```
Comment (bradrydzewski) [inline core/event.go]: Instead of creating a new event type, this value should be stored in the `Action` field.

**Comment by amq** [comment]
Could be behind a config variable.

**Comment by ashwilliams1** [comment]
There is a proposal here that seems perfectly fine to me:
https://github.com/drone/drone/issues/1878#issuecomment-287598789

The additional action types could be opt-in and only executed when the trigger section is configured accordingly. For example:
 
```
trigger:
  pull_request:
  - declined
  - closed
  - merged
```

**Comment by tboerger** [comment]
@ashwilliams1 if you replace `pull_request` with `action` drone is already prepared :)

**Comment by andrewhowdencom** [comment]
I had a go at implementing this. It's not tested, but looking for feedback:

https://github.com/drone/drone/pull/2902

**Comment by ashwilliams1** [comment]
I am going to close this in favor of #2902 which is based on this pull request, but uses the existing `action` attribute instead of adding a new event type.

**Pull request closed without merging**